### PR TITLE
grpc: implement pause and resume commands

### LINKED ISSUE #245: implement pause/resume commands
# Description of problem

pause and resume commands are needed to fix https://github.com/kata-containers/runtime/issues/317

### PR DESCRIPTION
**Pull request by devimc**
`pause` command is to pause a container and `resume` command
to resume a paused container, both commands use the libcontainer
API.

fixes #245

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: protocols/grpc/agent.proto [hunk 1]
```diff
@@ -31,6 +31,8 @@ service AgentService {
 	rpc ListProcesses(ListProcessesRequest) returns (ListProcessesResponse);
 	rpc UpdateContainer(UpdateContainerRequest) returns (google.protobuf.Empty);
 	rpc StatsContainer(StatsContainerRequest) returns (StatsContainerResponse);
+	rpc PauseContainer(PauseContainerRequest) returns (google.protobuf.Empty);
```
Comment (jodh-intel) [inline protocols/grpc/agent.proto]: Should these 2 be moved down to the "misc" section I wonder?

**Comment by jshachm** [comment]
LGTM

File: protocols/grpc/agent.proto (hunk 1 above)
Comment (devimc) [reply protocols/grpc/agent.proto]: I'm not sure, update and stats are also commands, so I placed pause and resume here

**Comment by bergwolf** [comment]
lgtm!

File: grpc.go [hunk 2]
```diff
@@ -867,6 +867,30 @@ func (a *agentGRPC) StatsContainer(ctx context.Context, req *pb.StatsContainerRe
...
 }
 
+func (a *agentGRPC) PauseContainer(ctx context.Context, req *pb.PauseContainerRequest) (*gpb.Empty, error) {
+	c, err := a.sandbox.getContainer(req.ContainerId)
+	if err != nil {
+		return emptyResp, err
+	}
+
+	a.sandbox.Lock()
+	defer a.sandbox.Unlock()
```
Comment (sboeuf) [inline grpc.go]: why are we using locks here ? I don't get the use case for that

Comment (caoruidong) [reply grpc.go]: @devimc Hi, please check this.

**Pull request merged**